src/dbnode/storage/entry_blackbox_test.go

```diff
@@ -247,3 +249,58 @@ func TestReconciledOnIndexSeries(t *testing.T) {
 	require.Equal(t, uint64(0), e.(*Entry).Index)
 	closer.Close()
 }
+
+func TestEntryTryMarkIndexGarbageCollected(t *testing.T) {
```

Comment:
i'm guessing this test fails with the previous code?

Reply:
Yeah - the wrong entry's `IndexGarbageCollected` bool will be set to true